src/main/java/com/hms_networks/americas/sc/extensions/json/JSONObject.java

```diff
@@ -809,6 +824,30 @@ public JSONObject put(String key, boolean value) throws JSONException {
     return this;
   }
 
+  /**
+   * Put a non-null key/boolean pair in the JSONObject.
+   *
+   * <p>This method removes the {@link JSONException} thrown by {@link #put(String, boolean)} in
+   * favor of a cleaner solution (where applicable).
+   *
+   * <p>It is strongly recommended that the standard {@link #put(String, boolean)} method be used
+   * when any possibility of a null key exists, since it is entirely up to the calling method to
+   * ensure that the key is not null and the value is not invalid (if applicable).
+   *
+   * @param key A key string (cannot be null).
+   * @param value A boolean which is the value.
+   * @return this.
+   * @since 1.15.0
+   */
+  public JSONObject putNonNull(String key, boolean value) {
+    // #if CLDC!="1.0"
+    putNonNull(key, value ? Boolean.TRUE : Boolean.FALSE);
+    // #else
+    // #         put(key, value ? TRUE : FALSE);
```

Comment:
the spacing looks odd here. I've seen other preprocessor directives in java where there is no space before the hashtag, does this still work?

Also, the spacing before put looks odd.

Reply:
I'm hesitant to change this formatting, as it matches the already established formatting for directives throughout the file. This file was adopted from public source JSON classes/files.

Reply:
sounds like it probably works as is.

Reply:
This whitespace is consistent with the existing conventions in this file as-is. 

Additional changes to the formatting here should be regarded as out-of-scope, and any concerns may be better addressed in a separate PR/issue.

Reply:
I've added a new [issue](https://github.com/hms-networks/sc-ewon-flexy-extensions-lib/issues/63) to look into this later.
It's fine as is for now.